.travis.yml

```diff
@@ -8,12 +8,13 @@ services:
  - docker
 
 before_script:
- - git diff --name-only --exit-code
  - docker build -t studio-frontend --no-cache .
 
 script:
   - npm run lint
   - npm run test
+  - make i18n.pre_validate
+  - make package-lock.validate
```

Comment:
The downside to this is that you won't know until after the full ~5 minutes that this is why the build failed. If it is in before_script then it will be run before anything else and will immediately fail the build without running any tests.

Reply:
That's interesting - I had moved it here because I noticed that even if these commands fail, the other commands around them will continue, as seen here: https://travis-ci.org/edx/studio-frontend/builds/338187324

This was helpful in the PR specifically, as I had all of linting, test, and i18n.pre_validate errors at various times and I was able to work on them in parallel. Should we prefer fast, sequential failures or longer, parallel full test suite runs?

Also worth pointing out - in the link above, the full run took 2:35, 1:14 of which was spent running `docker build -t studio-frontend --no-cache .`. Do we really need to continue using docker inside Travis? None of the other test commands make use of it.

Reply:
Yeah that's fair, I guess it's better to have the full test results.

If the tests work without the docker container, then I think we should do away with that line. I can't think of any reason we would need it. We could always add it back if needed.